using-mobile-extensions/offer-decisioning/api-reference.md

```diff
@@ -211,6 +211,160 @@ OfferDecisioning.retrievePrefetchedOffers(decisionScopes: [homeDecisionScope]) {
 ## Public classes
 
 {% tabs %}
+{% tab title="Android" %}
+### DecisionScope
+
+This class encapsulates the decision `activityId`, `placementId` and `itemCount`. The `DecisionScope` object can be used to fetch offers from the offer decisioning service if it is enabled in the Experience Edge network. The default value for `itemCount` is 1, if not provided when creating the `DecisionScope` object.
+
+```java
+final public class DecisionScope {
+    final private String activityId;
+    final private String placementId;
+    final private int itemCount;
```

Comment:
We can keep only the public properties/methods in this documentation - same applies for the classes below.

Reply:
I think for public classes it makes sense to indicate the properties they encapsulate. Especially when the extension's public API callback method is invoked with that custom object.

Reply:
In the long run, having developer-focused docs will help explain these public classes better.

Reply:
I am not sure the internal/private implementation details bring any benefit to the reader, the constructor and public setters/getters should be sufficient to indicate what parameters are required and what they represent. I see these these class/API documentation pages as public javadocs, where only the public functions/properties are documented. What do you think?

Reply:
Two things here regarding the class attributes/ properties:
1. Since Java doesn't explicitly mark convenience constructors, it's not easy to know the class properties unless getters (and setters) are inspected. Including those provides an easy reference.
2. Asymmetry: In Swift where the class properties may be defined as `public let` we shall include it in the docs here. 
Since these custom objects represent data that's returned to customers via callbacks, I believe it makes sense to include the class attributes.

Reply:
As discussed offline, let's keep this consistent with other open-source projects and shield the docs from internal implementation changes. Incorporated feedback.